pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -109,50 +106,49 @@ func getConsoleTemplate(cr *v1.KieApp) v1.ConsoleTemplate {
 // Returns the templates to use depending on whether the spec was defined with a common configuration
 // or a specific one.
 func getServersConfig(cr *v1.KieApp, commonConfig *v1.CommonConfig) ([]v1.ServerTemplate, error) {
-	servers := []v1.ServerTemplate{}
-	if cr.Spec.Objects.Servers != nil && len(cr.Spec.Objects.Servers) > 0 {
-		return servers, errors.New("invalid spec: provide either server or servers object")
-	}
-	if len(cr.Spec.Objects.Servers) != 0 {
-		for i, server := range cr.Spec.Objects.Servers {
-			kieServerID := fmt.Sprintf(defaultKieServerIDTemplate, cr.Name, i)
-			if len(server.Name) > 0 {
-				kieServerID = server.Name
-			}
-			crTemplate := v1.ServerTemplate{
-				KieServerID: kieServerID,
-			}
-			crTemplate.Build = getBuildConfig(commonConfig, server.Build)
-			if server.Build != nil {
-				crTemplate.From = getKieServerImageForBuild(commonConfig, i)
```

Comment:
@ruromero , we decided it doesn't make sense to have deployments > 1, when it's a build, right? Am I then correct to assume this should not have been there?